### PR TITLE
ACS-3572: Endpoint definition for GET action constraints (+model)

### PR DESCRIPTION
**Pull request by mpichura**

### CODE REVIEW

File: src/main/webapp/definitions/alfresco-core.yaml [hunk 1]
```diff
@@ -11027,6 +11065,45 @@ definitions:
          format: date-time
       values:
         type: object
+  ActionConstraint:
+    description: |
+      Response object holding action parameter constraint name and map of its values-labels.
```
Comment (tpage-alfresco) [inline src/main/webapp/definitions/alfresco-core.yaml:11070]: I assume the map is from labels to values?

File: src/main/webapp/definitions/alfresco-core.yaml [hunk 2]
```diff
@@ -8333,6 +8339,38 @@ paths:
           description: Unexpected error
           schema:
             $ref: '#/definitions/Error'
+  '/action-constraints':
```
Comment (tpage-alfresco) [inline src/main/webapp/definitions/alfresco-core.yaml:8342]: I wondered about changing this to `GET /action-constraints/{constraintName}`?

File: src/main/webapp/definitions/alfresco-core.yaml [hunk 3]
```diff
@@ -11027,6 +11065,45 @@ definitions:
...
+      Response object holding action parameter constraint name and map of its values-labels.
+    type: object
+    properties:
+      name:
+        type: string
+        description: |
+          Name of the constraint.
+      constraintsMap:
+        description: |
+          A map (String-String) of constraint values and labels.
```
Comment (tpage-alfresco) [inline src/main/webapp/definitions/alfresco-core.yaml:11079]: Do you happen to have an example of how this works in the v0 API?  I can't think how we can handle all the different constraints using a String-String map.

File: src/main/webapp/definitions/alfresco-core.yaml (hunk 2 above)
Comment (mpichura) [reply src/main/webapp/definitions/alfresco-core.yaml:8342]: Actually, I that may be more suitable. I am trying to figure out a reason when we would want to use GET all constraints and it seems that we would rather need GET single by name.

File: src/main/webapp/definitions/alfresco-core.yaml (hunk 3 above)
Comment (mpichura) [reply src/main/webapp/definitions/alfresco-core.yaml:11079]: This one isn't very intuitive. Constraint value is a key of the map, whereas the label is the value of the map. Constraint values are fixed and labels are translatable. See the email I sent you with example responses - maybe that will help a bit.

File: src/main/webapp/definitions/alfresco-core.yaml (hunk 1 above)
Comment (mpichura) [reply src/main/webapp/definitions/alfresco-core.yaml:11070]: See my other comment regarding that map.

**Review by krdabrowski — APPROVED** [review]

**Comment by mpichura** [comment]
After talking to Jamal, we decided to remove GET all constraints endpoint.

**Pull request merged**